File: app/src/main/java/kz/shymkent/relaxhouse/SharedPreferencesTools.java
package kz.shymkent.relaxhouse;

import android.content.Context;
import android.content.SharedPreferences;

import kz.shymkent.relaxhouse.application.App;

public class SharedPreferencesTools {
    static SharedPreferences sharedPreferences = App.getInstance().getSharedPreferences(Constants.APPLICATION_NAME, Context.MODE_PRIVATE);
    static SharedPreferences.Editor editor = sharedPreferences.edit();

    public static void savePhoneNumber(String phoneNumber){
        editor.putString(Constants.PHONE_NUMBER,phoneNumber);
        editor.commit();
    }
    public static String getPhoneNumber(){
        return sharedPreferences.getString(Constants.PHONE_NUMBER,"");
    }
    public static void saveCottageName(String cottageName){
        editor.putString(Constants.COTTAGE_NAME,cottageName);
        editor.commit();
    }
    public static String getCottagename(){
        return sharedPreferences.getString(Constants.COTTAGE_NAME,"");
    }

    public static void saveAuthorized(boolean authorized){
        editor.putBoolean(Constants.AUTHORIZED,authorized);
        editor.commit();
    }
    public static Boolean getAuthorized(){
        return sharedPreferences.getBoolean(Constants.AUTHORIZED,false);
    }

}

File: app/src/main/java/kz/shymkent/relaxhouse/SharedPreferencesTools.kt
package kz.shymkent.relaxhouse

import android.content.Context
import kz.shymkent.relaxhouse.application.App

object SharedPreferencesTools {
    private val sharedPreferences = App.instance.getSharedPreferences(Constants.APPLICATION_NAME, Context.MODE_PRIVATE)
    private val editor = sharedPreferences.edit()
    @JvmStatic
    fun savePhoneNumber(phoneNumber: String?) {
        editor.putString(Constants.PHONE_NUMBER, phoneNumber)
        editor.commit()
    }

    @JvmStatic
    val phoneNumber: String?
        get() = sharedPreferences.getString(Constants.PHONE_NUMBER, "")

    @JvmStatic
    fun saveCottageName(cottageName: String?) {
        editor.putString(Constants.COTTAGE_NAME, cottageName)
        editor.commit()
    }

    @JvmStatic
    val cottageName: String?
        get() = sharedPreferences.getString(Constants.COTTAGE_NAME, "")

    @JvmStatic
    fun saveAuthorized(authorized: Boolean) {
        editor.putBoolean(Constants.AUTHORIZED, authorized)
        editor.commit()
    }

    @JvmStatic
    val authorized: Boolean
        get() = sharedPreferences.getBoolean(Constants.AUTHORIZED, false)

}